### PR TITLE
Tickets/dm 20041

### PR DESCRIPTION
**Pull request by menanteau**

### CODE REVIEW

File: bin/get_topic_content [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    FORMAT_DATE = '%Y-%m-%d %H:%M:%S'
+    formatter = logging.Formatter(FORMAT, FORMAT_DATE)
+    logging.basicConfig(level=logging.NOTSET,
+                        format=FORMAT,
+                        datefmt=FORMAT_DATE)
+
+    args = cmdline()
+    for topic_name in args.topics:
+
+        SALconn = salpylib.DDSSubcriber(args.Device, topic_name, Stype=args.ctype, tsleep=0.1)
```
Comment (srp3rd) [inline bin/get_topic_content]: Minor nitpick:  Shouldn't this be "DDSSuscriber" ?

File: bin/get_topic_content [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        SALconn.start()
+        time.sleep(args.waittime)  # Give it some time to catchup
+        myData = SALconn.getCurrent(getNone=True)
+        print("There are: {} mesaage(s) stored".format(len(SALconn.myDatalist)))
```
Comment (srp3rd) [inline bin/get_topic_content]: Another minor thing: "messages"

**Review by srp3rd — APPROVED** [review]
This all looks fine.  Be aware that they made a change so that the last event will be retained by DDS in it's history cache, which is set to size=1.  That last event can't be cleared under the current version of SAL.

**Comment by timj** [comment]
Does @r-owen tackle this in salobj?

**Comment by r-owen** [comment]
By design SAL keeps one item of history for late joiners. ts_salobj 4 (the dds version) handles things a bit differently: by default salobj reads one item of history for events and telemetry, but you can read no history or more than 1 item by specifying max_history when you create the topic (though how many items are available depends on the quality of service, and that has to match what SAL uses). salobj 4reads no history for commands and command replies (unlike SAL).

File: bin/get_topic_content (hunk 1 above)
Comment (menanteau) [reply bin/get_topic_content]: Thanks for noticing.

File: bin/get_topic_content (hunk 2 above)
Comment (menanteau) [reply bin/get_topic_content]: Thanks for catching that.

**Pull request merged**